src/directives/mmsSearch.js

```diff
@@ -283,33 +348,40 @@ function mmsSearch($window, CacheService, ElementService, ProjectService, UtilsS
          * @param {number} page page number of search results
          * @param {number} numItems number of items to return per page
          */
-        scope.search = function(query, page, numItems) {
-            scope.searchClass = "fa fa-spin fa-spinner";
+        scope.search = function (query, page, numItems) {
+            scope.searchLoading = true;
             var queryOb = buildQuery(query);
-            queryOb.from = page*numItems + page;
+            queryOb.from = page * numItems + page;
             queryOb.size = numItems;
-            var reqOb = {projectId: scope.mmsProjectId, refId: scope.refId, checkType: true};
+            var reqOb = {
+                projectId: scope.mmsProjectId,
+                refId: scope.refId,
+                checkType: true
+            };
             ElementService.search(reqOb, queryOb, 2)
-            .then(function(data) {
-                if (scope.mmsOptions.filterCallback) {
-                  scope.searchResults = scope.mmsOptions.filterCallback(data);
-                } else {
-                  scope.searchResults = data;
-                }
-                scope.searchClass = "";
-                scope.currentPage = page;
-                scope.paginationCache.push(scope.searchResults);
-                if (scope.advanceSearch) {
-                    // scope.advanceSearch = !scope.advanceSearch;
-                    scope.advancedSearchResults = true;
-                }
-            }, function(reason) {
-                growl.error("Search Error: " + reason.message);
-                scope.searchClass = "";
-            });
+                .then(function (data) {
+                    if (scope.mmsOptions.filterCallback) {
+                        scope.searchResults = scope.mmsOptions.filterCallback(data);
+                        baseSearchResults = scope.searchResults;
+                    } else {
+                        scope.searchResults = data;
+                        baseSearchResults = data;
+                    }
+                    scope.currentPage = page;
+                    scope.paginationCache.push(scope.searchResults);
+                    if (scope.advanceSearch) {
+                        // scope.advanceSearch = !scope.advanceSearch;
+                        scope.advancedSearchResults = true;
+                    }
+                    scope.refineOptions = findRefineOptions(data);
```

Comment:
should pass baseSearchResults to findRefineOptions since filterCallback might filter data

Reply:
good catch